activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -150,6 +152,14 @@ def decryption_options
         def clean_text_scheme
           @clean_text_scheme ||= ActiveRecord::Encryption::Scheme.new(downcase: downcase?, encryptor: ActiveRecord::Encryption::NullEncryptor.new)
         end
+
+        def reserialize(value)
```

Comment:
Rather than doing this here, we could have an `EncryptedBinaryAttributeType` that subclasses ::ActiveModel::Type::Binary` instead and extract the common behaviour to a module.